[CI] Failure in org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.testRetentionLeasesClearedOnRestore

This failed in https://scans.gradle.com/s/atkkyov5t7yw4/console-log


```
REPRODUCE WITH: ./gradlew :plugins:repository-s3:testRepositoryCreds --tests "org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.testRetentionLeasesClearedOnRestore" -Dtests.seed=DD7008D9B5D2C3F5 -Dtests.security.manager=true -Dtests.locale=ar-LY -Dtests.timezone=America/Indiana/Knox -Dcompiler.java=12 -Druntime.java=11
```

Failed to get an ack for the close index call in that test:

```
java.lang.AssertionError: CloseIndexResponse failed - not acked
Expected: <true>
but: was <false>
```

Not sure this is necessarily related to snapshotting. I can reproduce it locally at a low rate though so I'm looking into it.

--- a/server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java
+++ b/server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java
@@ -442,7 +442,7 @@ public class MetaDataIndexStateService {
             final boolean acknowledged = result.getValue().hasFailures() == false;
             try {
                 if (acknowledged == false) {
-                    logger.debug("verification of shards before closing {} failed", index);
+                    logger.debug("verification of shards before closing {} failed [{}]", index, result);
                     continue;
                 }
                 final IndexMetaData indexMetaData = metadata.getSafe(index);

--- a/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java
@@ -311,6 +311,8 @@ public abstract class ESBlobStoreRepositoryIntegTestCase extends ESIntegTestCase
             addRandomDocuments(indexName, extraDocCount);
         }
 
+        // Wait for green so the close does not fail in the edge case of coinciding with a shard recovery that hasn't fully synced yet
+        ensureGreen();
         logger.info("-->  close index {}", indexName);
         assertAcked(client().admin().indices().prepareClose(indexName));
 

